sql/2021/pwa/top_manifest_display_values.sql

```diff
@@ -0,0 +1,58 @@
+#standardSQL
+# Top most used display values in manifest files
+
+CREATE TEMP FUNCTION getDisplay(manifest STRING) RETURNS STRING LANGUAGE js AS '''
+try {
+  var $ = Object.values(JSON.parse(manifest))[0];
+  if (!('display' in $)) {
+    return '(not set)';
+  }
+  return $.display;
+} catch {
+  return '(not set)'
+}
+''';
+
+SELECT
+  "PWA Sites" AS type,
+  _TABLE_SUFFIX AS client,
+  getDisplay(JSON_EXTRACT(payload, '$._pwa.manifests')) AS display,
+  COUNT(0) AS freq,
+  SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS total,
+  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS pct
+FROM
+  `httparchive.pages.2021_07_01_*`
+WHERE
+  JSON_EXTRACT(payload, '$._pwa') != "[]" AND
+  JSON_EXTRACT(payload, '$._pwa.manifests') != "[]" AND
+  JSON_EXTRACT(payload, '$._pwa.serviceWorkerHeuristic') = "true"
+GROUP BY
+  client,
+  display
+HAVING
+  display IS NOT NULL AND
+  freq > 100
+UNION ALL
+SELECT
+  "All Sites" AS type,
+  _TABLE_SUFFIX AS client,
+  getDisplay(JSON_EXTRACT(payload, '$._pwa.manifests')) AS display,
+  COUNT(0) AS freq,
+  SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS total,
+  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS pct
+FROM
+  `httparchive.pages.2021_07_01_*`
+WHERE
+  JSON_EXTRACT(payload, '$._pwa') != "[]" AND
+  JSON_EXTRACT(payload, '$._pwa.manifests') != "[]"
+GROUP BY
+  client,
+  display
+HAVING
+  display IS NOT NULL AND
+  freq > 1000
```

Comment:
Is this intentionally different from 100 above?

Reply:
Nope. Changed. There only are 5 in our current data set anyway so 100 definitely enough 😁